pkg/kubelet/pod_workers.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"sync"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/record"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/dockertools"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/golang/glog"
+)
+
+type podWorkers struct {
+	podsUpdates map[types.UID]chan api.BoundPod
+	dockerCache dockertools.DockerCache
+	syncPodFun  func(*api.BoundPod, dockertools.DockerContainers) error
+
+	// Used for draining the workers - should be used only in tests.
+	waitGroup *sync.WaitGroup
+}
+
+func newPodWorkers(dockerCache dockertools.DockerCache, syncPodFun func(*api.BoundPod, dockertools.DockerContainers) error) *podWorkers {
+	return &podWorkers{
+		podsUpdates: map[types.UID]chan api.BoundPod{},
+		dockerCache: dockerCache,
+		syncPodFun:  syncPodFun,
+	}
+}
+
+func (p *podWorkers) managePodLoop(podUpdates <-chan api.BoundPod) {
+	for newPod := range podUpdates {
+		// We may consider passing containers together with the pod from Kubelet.
+		// However, since we currently have cache this shouldn't cause performance
+		// overhead on Docker. Moreover, as long as we run syncPod no matter if
+		// it changes anything, having an old version of "containers" can cause
+		// starting unneded containers.
+		// TODO(wojtek-t): Revisit it once probing is divided from syncPod.
+		containers, err := p.dockerCache.RunningContainers()
+		if err != nil {
+			glog.Errorf("Error listing containers: %v", err)
+			p.doneTask()
+			continue
+		}
+		err = p.syncPodFun(&newPod, containers)
+		if err != nil {
+			glog.Errorf("Error syncing pod, skipping: %v", err)
```

Comment:
Sorry I left out the last part :) "syncing pod <UID>", lets add the pod's UID to the error